invokeai/backend/model_management/README.md

```diff
@@ -0,0 +1,27 @@
+# Model Cache
+
+## `glibc` Memory Allocator Fragmentation
+
+Python (and PyTorch) relies on the memory allocator from the C Standard Library (`libc`). On linux, with the GNU C Standard Library implementation (`glibc`), our memory access patterns have been observed to cause severe memory fragmentation. This fragmentation results in large amounts of memory that has been freed but can't be released back to the OS. Loading models from disk and moving them between CPU/CUDA seem to be the operations that contribute most to the fragmentation. This memory fragmentation issue can result in OOM crashes during frequent model switching, even if `max_cache_size` is set to a reasonable value (e.g. a OOM crash with `max_cache_size=16` on a system with 32GB of RAM).
+
+This problem may also exist on other OSes, and other `libc` implementations. But, at the time of writing, it has only been investigated on linux with `glibc`.
+
+To better understand how the `glibc` memory allocator works, see these references:
+- Basics: https://www.gnu.org/software/libc/manual/html_node/The-GNU-Allocator.html
+- Details: https://sourceware.org/glibc/wiki/MallocInternals
+
+Note the differences between memory allocated as chunks in an arena vs. memory allocated with `mmap`. Under `glibc`'s default configuration, most model tensors get allocated as chunks in an arena making them vulnerable to the problem of fragmentation.
+
+We can work around this memory fragmentation issue by setting the following env var:
+
+```bash
+# Force blocks >1MB to be allocated with `mmap` so that they are released to the system immediately when they are freed.
+MALLOC_MMAP_THRESHOLD_=1048576
```

Comment:
@lstein @hipsterusername  This is currently buried where no one will see it. I'm open to suggestions on where to put it.

A few questions:
1. Should this be the default or just a suggestion in the documentation? It could have unanticipated consequences, or behave differently across different `libc` implementations.
2. If we do make it the default. Where's the best place to put that? I'm not well-versed in all of the ways that users configure and launch the app today.

Reply:
We could just put `os.environ["MALLOC_MMAP_THRESHOLD_"]="1048576"` at or near the top of `api_app.py`

Reply:
Based on the discussion here: https://discord.com/channels/1020123559063990373/1149513695567810630/1159519282221285497 , and in private channels, I have:
- Made `MALLOC_MMAP_THRESHOLD_=1048576` the default in `invoke.sh`.
- Added a note in the manual installation docs for users who likely don't use `invoke.sh`.